### PR TITLE
feature(Jwt): Make cookie from jwtFilter conditional

### PR DESCRIPTION
**Pull request by gluizbr**

### CODE REVIEW

**Review by marcus-secato-symphony — APPROVED** [review]

**Review by symphony-thibault — COMMENTED** [review]
Hi @gluizbr, could you maybe document this in https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/tree/master/symphony-bdk-legacy/symphony-bdk-bot-sdk-java#configuration? 

As well, would it be possible to add a unit test that check that the `JwtCookieFilter` is not injected in the Spring context when prop `jwt-cookie.enable` is set to `true`?

**Comment by gluizbr** [comment]
> Hi @gluizbr, could you maybe document this in https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/tree/master/symphony-bdk-legacy/symphony-bdk-bot-sdk-java#configuration?
> 
> As well, would it be possible to add a unit test that check that the `JwtCookieFilter` is not injected in the Spring context when prop `jwt-cookie.enable` is set to `true`?

I've updated the documentation and added some UT to check if the JwtCookieFilter is disabled when the property is false

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/main/java/com/symphony/bdk/bot/sdk/webapi/security/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,94 +1,95 @@
...
+    registrationBean.setFilter(new RequestOriginFilter(
+        env.getProperty("request-origin.origin-header")));
+    registrationBean.addUrlPatterns(
+        env.getProperty("request-origin.url-mapping"));
+
+    return registrationBean;
+  }
+
+  @Bean
+  @ConditionalOnProperty(prefix = "jwt-cookie", name = "enable", havingValue = "true", matchIfMissing = true)
```
Comment (marcus-secato-symphony) [inline symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/main/java/com/symphony/bdk/bot/sdk/webapi/security/config/SecurityConfig.java:74]: matchIfMissing = false as we do not want it enabled by default

**Review by marcus-secato-symphony — CHANGES_REQUESTED** [review]

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/README.md [hunk 2]
```diff
@@ -120,6 +120,8 @@ Symphony Bot SDK requires two main configuration files, ```application.yaml``` (
 
 It is required to set the path to ```bot-config.json``` in ```application.yaml``` through the ```bot-config``` property, as well as, you must set the path to the RSA private key using ```certs``` property.
 
+It's posible to disable JwtCookie on authentication endpoint adding the jwt-cookie enable property
```
[the remark this answers is not in the document]
Comment (gluizbr) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/README.md]: done

File: symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/main/java/com/symphony/bdk/bot/sdk/webapi/security/config/SecurityConfig.java (hunk 1 above)
Comment (gluizbr) [reply symphony-bdk-legacy/symphony-bdk-bot-sdk-java/src/main/java/com/symphony/bdk/bot/sdk/webapi/security/config/SecurityConfig.java]: done

**Review by symphony-thibault — APPROVED** [review]
LGTM 👍

**Review by marcus-secato-symphony — APPROVED** [review]

**Pull request merged**